types/express-serve-static-core/index.d.ts

```diff
@@ -96,16 +98,22 @@ export type RequestHandlerParams<
     | ErrorRequestHandler<P, ResBody, ReqBody, ReqQuery, Locals>
     | Array<RequestHandler<P> | ErrorRequestHandler<P>>;
 
-export type RouteParameterNames<Route extends string> =
-    string extends Route
-        ? string
-        : Route extends `${string}:${infer Param}/${infer Rest}`
-        ? (Param | RouteParameterNames<Rest>)
-        : (Route extends `${string}:${infer LastParam}` ? LastParam : never);
-
-export type RouteParameters<T extends string> = {
-    [key in RouteParameterNames<T>]: string
-};
+type GetRouteParameter<RouteAfterColon extends string> = RouteAfterColon extends `${infer Char}${infer Rest}`
+    ? Char extends '/' | '-' | '.'
+        ? ''
+        : `${Char}${GetRouteParameter<Rest>}`
+    : RouteAfterColon;
+
+export type RouteParameters<Route extends string> = string extends Route
+    ? ParamsDictionary
+    : Route extends `${string}(${string}`
+    ? ParamsDictionary //TODO: handling for regex parameters
+    : Route extends `${string}:${infer Rest}`
+    ? (GetRouteParameter<Rest> extends `${infer ParamName}?`
+          ? { [P in ParamName]?: string }
+          : { [P in GetRouteParameter<Rest>]: string }) &
+          (Rest extends `${GetRouteParameter<Rest>}${infer Next}` ? RouteParameters<Next> : never)
+    : {};
```

Comment:
Maybe this should also be indented to improve readability, what do you think? 🙂 
Also, there are 5 spaces in lines 113 to 115.

```suggestion
export type RouteParameters<Route extends string> = string extends Route
    ? ParamsDictionary
    : Route extends `${string}(${string}`
        ? ParamsDictionary //TODO: handling for regex parameters
        : Route extends `${string}:${infer Rest}`
            ? (GetRouteParameter<Rest> extends `${infer ParamName}?`
                ? { [P in ParamName]?: string }
                : { [P in GetRouteParameter<Rest>]: string }) &
                (Rest extends `${GetRouteParameter<Rest>}${infer Next}` ? RouteParameters<Next> : never)
            : { };
```

Reply:
prettier did this, not sure why it's adding 5 spaces. would you prefer that i `prettier-ignore` this block and change the indentation?

Reply:
I'd say it's up to you. It was just a suggestion to make the code more readable and easy to work on. In the end, it should be ok if we end up with whatever prettier does. 🙂